### PR TITLE
feat(hogql): person_id in cohort

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

I found it very annoying I couldn't just query for cohorts in HogQL. This would have helped debug a customer issue.

## Changes

Adds `IN COHORT` support, such as:

```sql
SELECT event FROM events WHERE person_id IN COHORT 'my cohort'
```

This actually converts it to 

```sql
SELECT event FROM events WHERE person_id IN cohort('my cohort')
```

Prior to resolving `cohort()`, we swap it out for a subquery, whose shape depends on the cohort. Static cohorts get a slightly simpler query.

You can search for cohorts by ID or by name.

## How did you test this code?

Wrote a test, used the UI.

### CODE REVIEW

File: posthog/hogql/printer.py [hunk 1]
```diff
@@ -499,15 +499,8 @@ def visit_call(self, node: ast.Call):
...
-            if len(node.args) != 1:
-                raise HogQLException(
-                    f"Chart function '{node.name}' expects exactly one argument. Passed {len(node.args)}"
-                )
-            sparkline = f"tuple('__hogql_chart_type', 'sparkline', 'results', {self.visit(node.args[0])})"
-            if isinstance(self.stack[-1], ast.Alias):
-                return sparkline
-            return f"{sparkline} AS sparkline"
+        elif node.name in HOGQL_FUNCTIONS:
+            raise HogQLException(f"Unexpected unresolved HOGQL_FUNCTION '{node.name}(...)'")
```
Comment (Twixes) [inline posthog/hogql/printer.py:503]:
```suggestion
            raise HogQLException(f"Unexpected unresolved HogQL function '{node.name}(...)'")
```

**Review by Twixes — COMMENTED** [review]
Love this, though with the addition of `IN COHORT '{cohort_name}'` I'm worried this is gonna be really brittle. Makes it super easy to break a bunch of insights unexpectedly by renaming a cohort, and that's a hard problem to solve. We'd have to either rewrite all insights with `IN COHORT`/cohort()` when such a cohort is renamed, or at least show how many insights are going to be affected – otherwise it will be almost impossible to rename cohorts with any confidence for larger users of HogQL.
Maybe we could leave the name matching out for now and instead surface the cohort ID in definition cards + the cohort page? That's either way some extra work to make this feature work for users, but less possible pain

**Comment by mariusandra** [comment]
Hmm, I actually see this brittleness as a feature. We're targeting people writing SQL who should be able to understand the difference between "find by alias" (flexibility) and "find by id" (stability).

When I'm playing around and making queries, I definitely prefer writing `... in cohort 'PostHog Team'`. Seems much friendlier than `... in cohort 98` and looking up the id every time.

No matter if we write `in cohort 98` or `in cohort 'PostHog Team'`, you can still delete the cohort and invalidate a lot of queries, so regardless of the discussion here, I'd like us to track which queries use which cohorts. We'll want to do the same for event/person properties and likely other random metadata. However I think this is not something we should implement in this PR 🤔.

**Comment by mariusandra** [comment]
What about allowing it, but showing a warning/notice when the user searches for a cohort by name? https://github.com/PostHog/posthog/pull/16134

**Review by thmsobrmlr — APPROVED** [review]
IMO good enough that we display a warning when the user is filtering cohorts by name.

**Comment by mariusandra** [comment]
Thanks! Going to merge this in for now, and then let's clear up the warning/error/quick-fix system in the followup.

**Pull request merged**